### PR TITLE
Several deps bump: Actix, Tokio, Rustls, AWS SDK et al

### PR DESCRIPTION
**Pull request by brainstorm**

### CODE REVIEW

**Review by mmalenic — COMMENTED** [review]
Okay, I think I've gotten all dep bumps. Wasn't as bad as I thought once I realised `rustls_pemfile` was doing pretty much the same thing as before.

**Comment by mmalenic** [comment]
@brainstorm I would request your review but I can't since it's your PR.

File: htsget-config/src/tls/mod.rs [hunk 1]
```diff
@@ -141,51 +144,39 @@ impl KeyPairScheme for Option<&TlsServerConfig> {
...
-      .map_err(|err| ParseError(format!("failed to parse private key: {}", err)))?
-    {
-      Some(rustls_pemfile::Item::RSAKey(key)) => return Ok(PrivateKey(key)),
-      Some(rustls_pemfile::Item::PKCS8Key(key)) => return Ok(PrivateKey(key)),
-      Some(rustls_pemfile::Item::ECKey(key)) => return Ok(PrivateKey(key)),
+    match read_one(&mut key_reader)? {
+      Some(Pkcs1Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      Some(Pkcs8Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      Some(Sec1Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      // Silently disregard unknown private keys.
```
Comment (brainstorm) [inline htsget-config/src/tls/mod.rs:156]: 🤔 ... not sure we want to be so silent here?

File: htsget-search/src/storage/data_server.rs [hunk 2]
```diff
@@ -356,7 +360,7 @@ mod tests {
       "/data".to_string(),
     );
     let server = formatter.bind_data_server().await.unwrap();
-    assert_eq!(formatter.get_addr(), server.local_addr());
+    assert_eq!(formatter.get_addr(), server.local_addr().unwrap());
```
Comment (brainstorm) [inline htsget-search/src/storage/data_server.rs:363]: I'd use `TestResult` instead of unwrapping in tests, but nevermind for now.

**Review by brainstorm — COMMENTED** [review]
Very nice @mmalenic, you didn't even need to touch `tls.rs` :-!

File: htsget-config/src/tls/mod.rs (hunk 1 above)
Comment (mmalenic) [reply htsget-config/src/tls/mod.rs:156]: I think it's okay. This is what `rustls_pemfile` does by default within the `certs` function. The .pem file can have multiple certs/keys in it, so it's possible that a user wants to have the same file contain the certificate and key. This means that it shouldn't be an error to read a certificate and key from the same file.

**Review by mmalenic — APPROVED** [review]

**Pull request merged**